# HG changeset patch
# User rk
# Date 1322745521 -3600
# Node ID 3df18e85af2c7eba6930d5e294be3f486ca78473
# Parent  e656e1c236f1aac284968a3a33060ed5f1d6dbc3
When creating a floating window we copy the accelerator table of its logical
parent frame. Otherwise all accelerators will suddenly stop working when the
floating window has the focus.

diff -r e656e1c236f1 -r 3df18e85af2c samples/aui/auidemo.cpp
--- a/samples/aui/auidemo.cpp	Thu Dec 01 09:47:11 2011 +0100
+++ b/samples/aui/auidemo.cpp	Thu Dec 01 14:18:41 2011 +0100
@@ -715,7 +715,7 @@
     options_menu->AppendSeparator();
     options_menu->AppendCheckItem(ID_AllowToolbarResizing, _("Allow Toolbar Resizing"));
     options_menu->AppendSeparator();
-    options_menu->Append(ID_Settings, _("Settings Pane"));
+    options_menu->Append(ID_Settings, _("Settings Pane\tF10"));
 
     wxMenu* notebook_menu = new wxMenu;
     notebook_menu->AppendRadioItem(ID_NotebookArtGloss, _("Glossy Theme (Default)"));
diff -r e656e1c236f1 -r 3df18e85af2c src/aui/floatpane.cpp
--- a/src/aui/floatpane.cpp	Thu Dec 01 09:47:11 2011 +0100
+++ b/src/aui/floatpane.cpp	Thu Dec 01 14:18:41 2011 +0100
@@ -68,6 +68,19 @@
 #endif
 
     SetExtraStyle(wxWS_EX_PROCESS_IDLE);
+
+    // copy the accelerator table from our logical parent frame to this window
+    wxFrame* frame = wxDynamicCast(parent, wxFrame);
+    if(frame)
+    {
+        wxMenuBar* menuBar = frame->GetMenuBar();
+        if(menuBar)
+        {
+            wxAcceleratorTable* accelTable = menuBar->GetAcceleratorTable();
+            if(accelTable && accelTable->IsOk())
+                SetAcceleratorTable(*accelTable);
+        }
+    }
 }
 
 wxAuiFloatingFrame::~wxAuiFloatingFrame()
